ResearchKit/Common/ORKScaleSliderView.m

```diff
@@ -117,6 +128,26 @@ - (IBAction)sliderValueChanged:(id)sender {
     [self setCurrentValue:newValue];
 }
 
+static const CGFloat kVerticalSliderHorizontalMargin = 48;
+
+-(CGFloat)sliderLayoutWidth {
+    // Use the delegate expected width, if available, or if we have no delegate, then use our own width.
+    // Subtract the left and right margins from the width, to give the appropriate slider width.
+    return (self.delegate ? self.delegate.sliderLayoutWidth : self.previousLayoutWidth)-(kVerticalSliderHorizontalMargin*2);
+}
+
+-(void)layoutSubviews {
+    [super layoutSubviews];
+    // This is currently only used by vertical sliders, but keep track of the width to implement the Protocol anyway
+    CGFloat currentLayoutWidth = self.delegate ? self.delegate.sliderLayoutWidth : self.bounds.size.width;
```

Comment:
Why is this code in layoutSubviews? Perhaps the delegate relationship ought to be the other way round (so we get told when the required width changes)?